rust/protocol/src/crypto.rs

```diff
@@ -61,36 +73,53 @@ pub fn aes_256_cbc_decrypt(ctext: &[u8], key: &[u8], iv: &[u8]) -> Result<Vec<u8
         .map_err(|_| SignalProtocolError::InvalidCiphertext)?)
 }
 
-pub fn hmac_sha256(key: &[u8], input: &[u8]) -> Result<[u8; 32]> {
+/// The statically-known size of the output of [hmac_sha256].
+pub const HMAC_OUTPUT_SIZE: usize = 32;
+
+/// Calculate the HMAC-SHA256 code over `input` using `key`.
+pub fn hmac_sha256(key: &[u8], input: &[u8]) -> Result<[u8; HMAC_OUTPUT_SIZE]> {
     let mut hmac = Hmac::<Sha256>::new_varkey(key).expect("HMAC-SHA256 should accept any size key");
     hmac.update(input);
     Ok(hmac.finalize().into_bytes().into())
 }
 
+/// Length of the MAC key used for [aes256_ctr_hmacsha256_encrypt] and
+/// [aes256_ctr_hmacsha256_decrypt].
+pub const MAC_KEY_LENGTH: usize = 10;
+
+/// Encrypt plaintext `msg` using [aes_256_ctr_encrypt] with `cipher_key`. Then generate a MAC from
+/// the resulting ciphertext with [hmac_sha256] using `mac_key`, and append that to the result.
+///
+/// Only the first [MAC_KEY_LENGTH] bytes of the MAC key are used.
 pub fn aes256_ctr_hmacsha256_encrypt(
     msg: &[u8],
-    cipher_key: &[u8],
+    cipher_key: &[u8; AES_256_KEY_SIZE],
     mac_key: &[u8],
 ) -> Result<Vec<u8>> {
     let ctext = aes_256_ctr_encrypt(msg, cipher_key)?;
     let mac = hmac_sha256(mac_key, &ctext)?;
-    let mut result = Vec::with_capacity(ctext.len() + 10);
+    let mut result = Vec::with_capacity(ctext.len() + MAC_KEY_LENGTH);
     result.extend_from_slice(&ctext);
-    result.extend_from_slice(&mac[..10]);
+    result.extend_from_slice(&mac[..MAC_KEY_LENGTH]);
     Ok(result)
 }
 
+/// Validate the MAC `mac_key` against the ciphertext `ctext`, then decrypt `ctext` using
+/// [aes_256_ctr_decrypt] with `cipher_key`.
```

Comment:
I think those terms are less well-known, so if we wanted to use them we'd still have to define them (though possibly at the module level instead of here). Given that these are currently fairly minimal docs I think I'd go for the more wordy version that uses more common terms. @cosmicexplorer, do you have an opinion one way or the other?

Reply:
I am personally completely unfamiliar with NaCL/libsodium (sadly!!) but I think it would definitely be good to use canonical terminology if it exists. I don't have the context to really make the decision here but I think having consistent terminology is very useful so I would be super open to describing "sealing" and "opening" in the module-level docs, in a future PR? I will add that to #285!